{
  "configurations": {
    "CVE_data_version": "4.0",
    "nodes": [
      {
        "cpe_match": [
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.0:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.1:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.2:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.3:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.4:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.5:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.2.6:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.0:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.1:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.2:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.3:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.4:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.5:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.3.6:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.4.0:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.4.1:*:*:*:*:*:*:*",
            "vulnerable": true
          },
          {
            "cpe23Uri": "cpe:2.3:a:python:python:3.4.2:*:*:*:*:*:*:*",
            "vulnerable": true
          }
        ],
        "operator": "OR"
      }
    ]
  },
  "cve": {
    "CVE_data_meta": {
      "ASSIGNER": "cve@mitre.org",
      "ID": "CVE-2014-2667"
    },
    "affects": {
      "vendor": {
        "vendor_data": [
          {
            "product": {
              "product_data": [
                {
                  "product_name": "python",
                  "version": {
                    "version_data": [
                      {
                        "version_affected": "=",
                        "version_value": "3.2.0"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.2.1"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.2.2"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.2.3"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.2.4"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.2.5"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.2.6"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.0"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.1"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.2"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.3"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.4"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.5"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.3.6"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.4.0"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.4.1"
                      },
                      {
                        "version_affected": "=",
                        "version_value": "3.4.2"
                      }
                    ]
                  }
                }
              ]
            },
            "vendor_name": "python"
          }
        ]
      }
    },
    "data_format": "MITRE",
    "data_type": "CVE",
    "data_version": "4.0",
    "description": {
      "description_data": [
        {
          "lang": "en",
          "value": "Race condition in the _get_masked_mode function in Lib/os.py in Python 3.2 through 3.5, when exist_ok is set to true and multiple threads are used, might allow local users to bypass intended file permissions by leveraging a separate application vulnerability before the umask has been set to the expected value."
        }
      ]
    },
    "problemtype": {
      "problemtype_data": [
        {
          "description": [
            {
              "lang": "en",
              "value": "CWE-362"
            }
          ]
        }
      ]
    },
    "references": {
      "reference_data": [
        {
          "name": "http://bugs.python.org/issue21082",
          "refsource": "CONFIRM",
          "tags": [
            "Vendor Advisory"
          ],
          "url": "http://bugs.python.org/issue21082"
        },
        {
          "name": "openSUSE-SU-2014:0596",
          "refsource": "SUSE",
          "tags": [],
          "url": "http://lists.opensuse.org/opensuse-updates/2014-05/msg00007.html"
        },
        {
          "name": "openSUSE-SU-2014:0597",
          "refsource": "SUSE",
          "tags": [],
          "url": "http://lists.opensuse.org/opensuse-updates/2014-05/msg00008.html"
        },
        {
          "name": "[oss-security] 20140328 CVE request: os.makedirs(exist_ok=True) is not thread-safe in Python",
          "refsource": "MLIST",
          "tags": [],
          "url": "http://www.openwall.com/lists/oss-security/2014/03/28/15"
        },
        {
          "name": "[oss-security] 20140329 Re: [PSRT] CVE request: os.makedirs(exist_ok=True) is not thread-safe in Python",
          "refsource": "MLIST",
          "tags": [],
          "url": "http://www.openwall.com/lists/oss-security/2014/03/29/5"
        },
        {
          "name": "[oss-security] 20140330 Re: CVE request: os.makedirs(exist_ok=True) is not thread-safe in Python",
          "refsource": "MLIST",
          "tags": [],
          "url": "http://www.openwall.com/lists/oss-security/2014/03/30/4"
        },
        {
          "name": "GLSA-201503-10",
          "refsource": "GENTOO",
          "tags": [],
          "url": "https://security.gentoo.org/glsa/201503-10"
        }
      ]
    }
  },
  "impact": {
    "baseMetricV2": {
      "cvssV2": {
        "accessComplexity": "MEDIUM",
        "accessVector": "LOCAL",
        "authentication": "NONE",
        "availabilityImpact": "NONE",
        "baseScore": 3.3,
        "confidentialityImpact": "PARTIAL",
        "integrityImpact": "PARTIAL",
        "vectorString": "AV:L/AC:M/Au:N/C:P/I:P/A:N",
        "version": "2.0"
      },
      "exploitabilityScore": 3.4,
      "impactScore": 4.9,
      "obtainAllPrivilege": false,
      "obtainOtherPrivilege": false,
      "obtainUserPrivilege": false,
      "severity": "LOW",
      "userInteractionRequired": false
    }
  },
  "lastModifiedDate": "2017-07-01T01:29Z",
  "publishedDate": "2014-11-16T01:59Z"
}